python/lsst/ts/mtaos/mtaos.py

```diff
@@ -342,6 +343,11 @@ async def configure(self, config: typing.Any) -> None:
 
         self.log.debug("ofc data ready. Creating model")
 
+        try:
+            self.zernike_column_pattern = re.compile(config.zernike_column_pattern)
+        except re.error as e:
+            raise salobj.ExpectedError(f"Failed to compile zernike_column_pattern with {e}.")
```

Comment:
It would be good if you add more information to the user. Also, instead of doing `f"... {e}."`, consider doing `from e`, e.g.:

```
raise salobj.ExpectedError(
    f"Failed to compile {config.zernike_columns_pattern=}. "
    "This means the valid defined for the zernike column patter is invalid. "
    "The value must be a valid Regular Expression. "
    "Contact support to get help updating the CSC configuration, or select a different configuration."
) from e
```

This will ensure the full traceback will be added to the exception you are raising.